pkg/products/threescale/reconciler.go

```diff
@@ -342,6 +383,260 @@ func (r *Reconciler) backupSystemSecrets(ctx context.Context, serverClient k8scl
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
 
+func getDeploymentConfig(ctx context.Context, client k8sclient.Client, dcName string) (*appsv1.DeploymentConfig, integreatlyv1alpha1.StatusPhase, error) {
+	apiCastDeploymentConfig := &appsv1.DeploymentConfig{}
+
+	err := client.Get(ctx, k8sTypes.NamespacedName{Name: dcName, Namespace: "redhat-rhmi-3scale"}, apiCastDeploymentConfig)
```

Comment:
```suggestion
func (r *Reconciler) getDeploymentConfig(ctx context.Context, client k8sclient.Client, dcName string) (*appsv1.DeploymentConfig, integreatlyv1alpha1.StatusPhase, error) {
	apiCastDeploymentConfig := &appsv1.DeploymentConfig{}

	err := client.Get(ctx, k8sTypes.NamespacedName{Name: dcName, Namespace: r.Config.GetProductNamespace()}, apiCastDeploymentConfig)
```

Change to avoid hardcoding the namespace. I'd suggest the same in the `getService` function